### PR TITLE
Feature/arc i preprocessor

### PR DESCRIPTION
**Pull request by pl8787**
#359

### CODE REVIEW

**Comment by bwanglzu** [comment]
will look into this a bit later

File: matchzoo/preprocessor/arci_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+"""ArcI Preprocessor."""
+
+from matchzoo import engine
+from matchzoo import preprocessor
+from matchzoo import datapack
+from matchzoo.embedding import Embedding
+
+import os
```
Comment (lixinsu) [inline matchzoo/preprocessor/arci_preprocessor.py]: python包的导入顺序：1. Python 标准库模块 2. Python 第三方模块3. 应用程序自定义模块
有助于减少每个模块需要的 import 语句数目。

File: matchzoo/preprocessor/arci_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+
+class ArcIPreprocessor(engine.BasePreprocessor):
+    """
+    ArcI preprocessor helper.
+
+    TODO: NEED REFACTORING.
```
Comment (faneshion) [inline matchzoo/preprocessor/arci_preprocessor.py]: More detailed information?

Comment (pl8787) [reply matchzoo/preprocessor/arci_preprocessor.py]: About what?

File: matchzoo/preprocessor/arci_preprocessor.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+        # Initialize a vocabulary process unit to build words vocab.
+        self._vocab_unit.fit(vocab)
+
+        if os.path.isfile(self._embedding_file):
+            embed_module = Embedding(embedding_file=self._embedding_file)
+            embed_module.build(self._vocab_unit.state['term_index'])
+            self._context['embedding_mat'] = embed_module.embedding_mat
+        else:
+            logger.info("Embedding file [{}] not found."
```
Comment (bwanglzu) [inline matchzoo/preprocessor/arci_preprocessor.py]: when can not find embedding file, we should raise an Exception.

**Pull request merged**